specutils/analysis/moment.py

```diff
@@ -43,10 +44,21 @@ def moment(spectrum, regions=None, order=0, axis=-1):
                                order=order, axis=axis)
 
 
-def _compute_moment(spectrum, regions=None, order=0, axis=-1):
+def _compute_moment(spectrum, regions=None, order=0, axis=None):
     """
     This is a helper function for the above `moment()` method.
     """
+    if axis is None:
+        if isinstance(spectrum, SpectrumCollection):
+            axes = [spec.spectral_axis_index for spec in spectrum]
+            if not np.all([x==axes[0] for x in axes]):
```

Comment:
super minor, but I wonder if this might be better stated as `if len(set(axes)) != 1:` ?   Performance-wise, these are probably the same, but this reads a little more clearly to me.

Reply:
I considered that and went with this way mostly at random. Happy to change it to use `set`.

Reply:
Really either way is fine.